src/controller.js

```diff
@@ -0,0 +1,126 @@
+        document.querySelector("#addbutton").disabled = true;
+      });
+    }
+
+    initialiseSea() {
+      const backgrounds = [
+        "./images/water0.png",
+        "./images/water1.png",
+      ];
+      let backgroundIndex = 0;
+      window.setInterval(() => {
+        document.querySelector("#viewport").style.backgroundImage = `url("${backgrounds[backgroundIndex % backgrounds.length]}")`;
+        backgroundIndex += 1;
+      }, 1000);
+    }
+
+    resetContainer(parent){
+      while (parent.firstChild) {
+          parent.removeChild(parent.firstChild);
+      }
+    }
+
+    renderPorts(ports) {
+      const portsElement = document.querySelector("#ports");
+      portsElement.style.width = "0px";
+      this.resetContainer(portsElement);
+      ports.forEach((port, index) => {
+        const newPortElement = document.createElement("div");
+        newPortElement.className = "port";
+        newPortElement.dataset.portName = port.name;
+        newPortElement.dataset.portIndex = index;
+        portsElement.appendChild(newPortElement);
+        const portsElementWidth = parseInt(portsElement.style.width, 10);
+        portsElement.style.width = `${portsElementWidth + 256}px`;
+      })
+    }
+
+    renderShip() {
+      const ship = this.ship;
+      const shipPortIndex = ship.itinerary.ports.indexOf(ship.currentPort);
+      const portElement = document.querySelector(`[data-port-index="${shipPortIndex}"]`);
+      if (!portElement) return;
+      const shipElement = document.querySelector("#ship");
+      shipElement.style.top = `${portElement.offsetTop + 32}px`;
+      shipElement.style.left = `${portElement.offsetLeft - 32}px`;
+    }
+
+    setSail() {
+      const ship = this.ship;
+      const currentPortIndex = ship.itinerary.ports.indexOf(ship.currentPort);
+      const nextPortIndex = currentPortIndex + 1;
+      const nextPortElement = document.querySelector(`[data-port-index="${nextPortIndex}"]`);  
+      if (!nextPortElement) {
+        this.renderMessage("End of the line!");
+        return;
+      } else {
+        this.renderMessage(`Now departing ${ship.currentPort.name}`);
+      }
+      const shipElement = document.querySelector("#ship");
+      const sailInterval = setInterval(() => {
+        const shipLeft = parseInt(shipElement.offsetLeft, 10);
+        if (shipLeft === (nextPortElement.offsetLeft - 32)) {
+          document.querySelector("#sailbutton").disabled = false;
+          ship.setSail();
+          ship.dock();
+          clearInterval(sailInterval);
+          this.renderMessage(`Now arriving at ${ship.currentPort.name}`);
+          if (currentPortIndex + 1 === ship.itinerary.ports.length -1) {
+            this.updateDisplay(`Current Port: ${ship.currentPort.name}`, `Next Port: N/A End of the line!`);
+          } else {
+            this.updateDisplay(`Current Port: ${ship.currentPort.name}`, `Next Port: ${ship.itinerary.ports[nextPortIndex + 1].name}`);
+          }
```

Comment:
I would move this functionality inside of `updateDisplay`, that way it would be easier to call it in different circumstances e.g. like I mentioned earlier, when you add a port while at the end of the line